src/script/miniscript.h

```diff
@@ -298,6 +298,8 @@ struct Node {
     const Type typ;
     //! Cached script length (computed by CalcScriptLen).
     const size_t scriptlen;
+    //! Whether a public key appears more than once in this node.
+    const bool duplicate_key;
```

Comment:
Naming could be improved. The name `duplicate_key` might suggest that the variable holds the value or reference of the key that is duplicated, when in fact it can only hold either `true` or `false`—and there could be multiple duplicated keys. It'd be best to name this variable `has_duplicate_keys` instead, which would go nicely along the naming scheme of `IsNonMalleable` and `IsSane`.

Reply:
> The name duplicate_key might suggest that the variable holds the value or reference of the key that is duplicated, when in fact it can only hold either true or false

I don't think it's worth making the variable name even longer. The `bool` type should be enough for one to get that this value holds, well, a boolean and not a reference to the key.
Also note this commit is part of https://github.com/bitcoin/bitcoin/pull/24860 which features a `ContainsDuplicateKey()` public API in the same spirit as `IsNonMalleable` and `IsSane`.

> and there could be multiple duplicated keys

The detection algorithm cuts through and wouldn't know that, it just knows that "there exists one duplicate key".